File: InternetRT/app/org/internetrt/driver/userinterface/BrowserBaseClientInputs.java
package org.internetrt.driver.userinterface;

import org.internetrt.CONSTS;
import org.internetrt.SiteUserInterface;

import play.mvc.Controller;
import play.mvc.Result;

public class BrowserBaseClientInputs extends Controller{

	
	public static Result register(){
		String username = request().queryString().get("username")[0];
		String password = request().queryString().get("password")[0];
		return ok(SiteUserInterface.register(username,password));
	}
	public static Result login(){
		String username = request().queryString().get("username")[0];
		String password = request().queryString().get("password")[0];
		
		try{
			System.out.println("111111111111111111111111111111111");
			String uid = SiteUserInterface.login(username,password);
			System.out.println("2222222222222222222222222222222222");
			session().put(CONSTS.SESSIONUID(), uid);
			System.out.println("33333333333333333333333333333333333");
			return ok(uid);
		}catch(Exception e){
			return ok(e.getMessage());
		}	
	}
	
	public static Result installRootApp(){
		String uid = session().get(CONSTS.SESSIONUID());
		String xml = request().queryString().get("xml")[0];
		
		Boolean success =  SiteUserInterface.installRootApp(uid,xml);
		
		return ok(success.toString());
	}
	
}

File: InternetRT/app/org/internetrt/driver/userinterface/BrowserBaseClientInputs.scala
package org.internetrt.driver.userinterface

import play.api.mvc._
import org.internetrt.SiteUserInterface
import org.internetrt.CONSTS
import org.internetrt.exceptions.ConsideredException

object BrowserBaseClientInputs extends Controller {
  def register() = Action {
    implicit request =>
      if (request.method == "GET") {
        Ok(views.html.register())
      } else {
        val username = request.body.asFormUrlEncoded.get("username").head;
        val password = request.body.asFormUrlEncoded.get("password").head;
		SiteUserInterface.register(username, password)
		val mainpage = controllers.routes.Application.index().absoluteURL(false)
        Ok(views.html.login(mainpage,"Register successed,please login"))
      }
  }
  def getName() = Action {
    request =>
      Ok(request.session.get("username").getOrElse(""))
  }
  def login() = Action {
    implicit request =>
      System.out.println("Before Login:" + request.session)

      val mainpage = controllers.routes.Application.index().absoluteURL(false)
      if (request.method == "GET") {
        val oldurl = request.queryString.get("oldurl").getOrElse(Seq.empty).headOption.getOrElse(mainpage);
        Ok(views.html.login(oldurl,""))
      } else {
        val username = request.body.asFormUrlEncoded.get("username").headOption;
        val password = request.body.asFormUrlEncoded.get("password").headOption;
        val oldurl = request.body.asFormUrlEncoded.get("oldurl").headOption.getOrElse(mainpage);

        try {
          val uid = SiteUserInterface.login(username.get, password.get);

          System.out.println("UID in login" + uid);
          System.out.println("oldurl:" + oldurl);
          Redirect(oldurl).withSession(request.session +
            (CONSTS.SESSIONUID -> uid) + ("username" -> username.get));
        } catch {
          case e:ConsideredException => {
            e.printStackTrace();
            Ok(views.html.login(mainpage,"Login failed,Please try again."))
          }
        }
      }
  }
  def confirmRouting() = Action {
    implicit request =>

      request.session.get(CONSTS.SESSIONUID) match {
        case Some(uid) => {
          request.body.asFormUrlEncoded.get("xml") match {
            case Seq(xml) => {

              val success = SiteUserInterface.confirmRouting(uid, xml);

              Ok(success.toString());
            }
            case _ => InternalServerError
          }
        }
        case None => {
          val thispage = controllers.routes.Application.index().absoluteURL(false)
          Ok(views.html.login(thispage,"login first"))
        }
      }
  }
  def installRootApp() = Action {
    implicit request =>

      System.out.println("Install root app" + request.session)

      if (request.method == "GET") Ok(views.html.installRootApp())
      else request.session.get(CONSTS.SESSIONUID) match {
        case Some(uid) => {
          request.body.asFormUrlEncoded.get("xml") match {
            case Seq(xml) => {
              System.out.println("UID" + uid);
              System.out.println("XML" + xml);

              val success = SiteUserInterface.installRootApp(uid, xml);

              Ok(success.toString());
            }
            case _ => InternalServerError
          }
        }
        case None => {
          val thispage = controllers.routes.Application.index().absoluteURL(false)
          Ok(views.html.login(thispage,"login first"))
        }
      };

  }

}